Sources/StylePart.swift

```diff
@@ -174,8 +191,33 @@ extension StringStyle {
                     self.fontFeatureProviders += [numberSpacing as FontFeatureProvider]
                     return
                 }
+                else if case let .superscript(superscript) = stylePart {
+                    self.fontFeatureProviders += [superscript ? VerticalPosition.superscript : VerticalPosition.normal as FontFeatureProvider]
+                    return
+                }
+                else if case let .`subscript`(`subscript`) = stylePart {
+                    self.fontFeatureProviders += [`subscript` ? VerticalPosition.`subscript` : VerticalPosition.normal as FontFeatureProvider]
+                    return
+                }
+                else if case let .ordinals(ordinals) = stylePart {
+                    self.fontFeatureProviders += [ordinals ? VerticalPosition.ordinals : VerticalPosition.normal as FontFeatureProvider]
+                    return
+                }
+                else if case let .scientificInferiors(scientificInferiors) = stylePart {
+                    self.fontFeatureProviders += [scientificInferiors ? VerticalPosition.scientificInferiors : VerticalPosition.normal as FontFeatureProvider]
+                    return
+                }
+                else if case let .smallCaps(smallCaps) = stylePart {
+                    self.fontFeatureProviders += [smallCaps as FontFeatureProvider]
+                    return
+                }
+                else if case let .stylisticAlternates(stylisticAlternates) = stylePart {
+                    self.fontFeatureProviders += [stylisticAlternates as FontFeatureProvider]
+                    return
+                }
                 else if case let .fontFeature(featureProvider) = stylePart {
                     self.fontFeatureProviders.append(featureProvider)
+                    return
```

Comment:
1) Why is there a return? I see it was already there, but it doesn't seem needed
2) The StringStyle object models these features in properties. They should be transferred into those properties, not into the fontFeatureProviders array. That array was sort of a generic bucket, I think we may way to look at it again.